go/vt/vttablet/tabletserver/throttle/mysql/mysql_throttle_metric.go

```diff
@@ -108,50 +111,53 @@ func GetMetricsQueryType(query string) MetricsQueryType {
 
 // MySQLThrottleMetric has the probed metric for a tablet
 type MySQLThrottleMetric struct { // nolint:revive
-	ClusterName string
-	Alias       string
-	Value       float64
-	Err         error
+	Name  base.MetricName
+	Scope base.Scope
+	Alias string
+	Value float64
+	Err   error
 }
 
+type MySQLThrottleMetrics map[base.MetricName]*MySQLThrottleMetric // nolint:revive
+
 // NewMySQLThrottleMetric creates a new MySQLThrottleMetric
 func NewMySQLThrottleMetric() *MySQLThrottleMetric {
 	return &MySQLThrottleMetric{Value: 0}
 }
 
 // GetClusterTablet returns the ClusterTablet part of the metric
-func (metric *MySQLThrottleMetric) GetClusterTablet() ClusterTablet {
-	return GetClusterTablet(metric.ClusterName, metric.Alias)
+func (metric *MySQLThrottleMetric) GetTabletAlias() string {
+	return metric.Alias
 }
 
 // Get implements MetricResult
 func (metric *MySQLThrottleMetric) Get() (float64, error) {
 	return metric.Value, metric.Err
 }
 
-// ReadThrottleMetric returns a metric for the given probe. Either by explicit query
+// WithError returns this metric with given error
+func (metric *MySQLThrottleMetric) WithError(err error) *MySQLThrottleMetric {
+	metric.Err = err
+	return metric
+}
+
+// ReadThrottleMetrics returns a metric for the given probe. Either by explicit query
 // or via SHOW REPLICA STATUS
-func ReadThrottleMetric(probe *Probe, clusterName string, overrideGetMetricFunc func() *MySQLThrottleMetric) (mySQLThrottleMetric *MySQLThrottleMetric) {
-	if mySQLThrottleMetric := getCachedMySQLThrottleMetric(probe); mySQLThrottleMetric != nil {
-		return mySQLThrottleMetric
-		// On cached results we avoid taking latency metrics
+func ReadThrottleMetrics(ctx context.Context, probe *Probe, metricsFunc func(context.Context) MySQLThrottleMetrics) MySQLThrottleMetrics {
+	if metrics := getCachedMySQLThrottleMetrics(probe); metrics != nil {
+		return metrics
 	}
 
 	started := time.Now()
-	mySQLThrottleMetric = NewMySQLThrottleMetric()
-	mySQLThrottleMetric.ClusterName = clusterName
-	mySQLThrottleMetric.Alias = probe.Alias
-
-	defer func(metric *MySQLThrottleMetric, started time.Time) {
-		go func() {
-			stats.GetOrNewGauge("ThrottlerProbesLatency", "probes latency").Set(time.Since(started).Nanoseconds())
-			stats.GetOrNewCounter("ThrottlerProbesTotal", "total probes").Add(1)
-			if metric.Err != nil {
-				stats.GetOrNewCounter("ThrottlerProbesError", "total probes errors").Add(1)
-			}
-		}()
-	}(mySQLThrottleMetric, started)
-
-	mySQLThrottleMetric = overrideGetMetricFunc()
-	return cacheMySQLThrottleMetric(probe, mySQLThrottleMetric)
+	mySQLThrottleMetrics := metricsFunc(ctx)
+
+	go func(metrics MySQLThrottleMetrics, started time.Time) {
+		stats.GetOrNewGauge("ThrottlerProbesLatency", "probes latency").Set(time.Since(started).Nanoseconds())
+		stats.GetOrNewCounter("ThrottlerProbesTotal", "total probes").Add(1)
+		// if metric.Err != nil {
+		// 	stats.GetOrNewCounter("ThrottlerProbesError", "total probes errors").Add(1)
+		// }
```

Comment:
nit: any reason to keep this as a comment?

Reply:
Good catch. Fixed (uncommented and adapted, now reporting the metric)